### PR TITLE
feat: section press events

### PR DESCRIPTION
**Pull request by tuncaulubilge**
Article press and puzzle press events on sections.

+ Also fixed puzzle click analytics

### CODE REVIEW

File: packages/section/src/section.js [hunk 1]
```diff
@@ -29,14 +29,14 @@ class Section extends Component {
...
+      onArticlePress,
+      onPuzzlePress,
       publicationName,
       section: { slices, title },
       onViewed,
       receiveChildList
     } = this.props;
-
-    const showSeparator = title !== "Puzzles";
+    const isPuzzle = title === "Puzzles";
```
Comment (aronshamash) [inline packages/section/src/section.js]: this is horrible but hey!

Comment (tuncaulubilge) [reply packages/section/src/section.js]: I'll replace this with a typename check

Comment (Ivaylo1987) [reply packages/section/src/section.js]: It was horrible even before :)

Comment (tuncaulubilge) [reply packages/section/src/section.js]: Replaced with a typename check, instead of a title match now @aronshamash

Comment (aronshamash) [reply packages/section/src/section.js]: nice

**Pull request merged**